eventlogger/src/main/java/gov/cms/ab2d/eventlogger/eventloggers/sql/ApiRequestEventMapper.java

```diff
@@ -3,12 +3,64 @@
 import gov.cms.ab2d.eventlogger.EventLoggingException;
 import gov.cms.ab2d.eventlogger.LoggableEvent;
 import gov.cms.ab2d.eventlogger.events.ApiRequestEvent;
+import gov.cms.ab2d.eventlogger.utils.UtilMethods;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.support.GeneratedKeyHolder;
+import org.springframework.jdbc.support.KeyHolder;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.time.OffsetDateTime;
+
+public class ApiRequestEventMapper extends SqlEventMapper {
+    private JdbcTemplate template;
+
+    public ApiRequestEventMapper(JdbcTemplate template) {
+        this.template = template;
+    }
 
-public class ApiRequestEventMapper implements SqlEventMapper {
     @Override
     public void log(LoggableEvent event) {
         if (event.getClass() != ApiRequestEvent.class) {
             throw new EventLoggingException("Used " + event.getClass().toString() + " instead of " + ApiRequestEvent.class.toString());
         }
+        String query = "insert into event_api_request " +
+                " (time_of_event, user_id, job_id, url, ip_address, token_hash, request_id) " +
+                " values (?, ?, ?, ?, ?, ?, ?)";
```

Comment:
Using named sql parameters would definitely be more readable

Reply:
Interesting. I will.